libraries/chain/asset_evaluator.cpp

```diff
@@ -393,22 +393,51 @@ void_result asset_update_bitasset_evaluator::do_evaluate(const asset_update_bita
    return void_result();
 } FC_CAPTURE_AND_RETHROW( (o) ) }
 
-void_result asset_update_bitasset_evaluator::do_apply(const asset_update_bitasset_operation& o)
-{ try {
-   bool should_update_feeds = false;
-   // If the minimum number of feeds to calculate a median has changed, we need to recalculate the median
-   if( o.new_options.minimum_feeds != bitasset_to_update->options.minimum_feeds )
-      should_update_feeds = true;
-
-   db().modify(*bitasset_to_update, [&](asset_bitasset_data_object& b) {
-      b.options = o.new_options;
+void_result asset_update_bitasset_evaluator::do_apply(const asset_update_bitasset_operation& op)
+{
+   try {
+      // If the minimum number of feeds to calculate a median has changed, we need to recalculate the median
+      bool should_update_feeds = false;
+      if( op.new_options.minimum_feeds != bitasset_to_update->options.minimum_feeds )
+         should_update_feeds = true;
+
+      // feeds must be reset if the backing asset is changed
+      bool backing_asset_changed = false;
+      bool is_witness_or_committee_fed = false;
+      if (op.new_options.short_backing_asset != bitasset_to_update->options.short_backing_asset)
+      {
+         backing_asset_changed = true;
+         const asset_object& base_asset = op.asset_to_update(db());
+         if ( base_asset.options.flags & (witness_fed_asset | committee_fed_asset) )
+            is_witness_or_committee_fed = true;
+      }
 
-      if( should_update_feeds )
-         b.update_median_feeds(db().head_block_time());
-   });
+      // now do the actual modifications to the database object
+      db().modify(*bitasset_to_update, [&](asset_bitasset_data_object& bdo) {
+         bdo.options = op.new_options;
+
+         // are we modifying the underlying? If so, reset the feeds
+         if (backing_asset_changed) {
+            if ( is_witness_or_committee_fed ) {
+               bdo.feeds.clear();
+            } else {
+               // for non-witness-feeding and non-committe-feeding assets, modify all feeds
+               // published by producers to nothing, since we can't simply remove them. For more information:
+               // https://github.com/bitshares/bitshares-core/pull/832#issuecomment-384112633
+               for(auto& current_feed : bdo.feeds) {
+                  // If this logic grows, it may be good to move it to a reset method in price_feed
+                  current_feed.second.second.settlement_price = price();
```

Comment:
Possible case: If price feeds are [nan, nan, 300], the median price feed is still nan. Is that to be expected?

Reply:
NaN will be filtered out when calculating median feed.

Reply:
I would think so too. But my testing shows otherwise. The relevant code:
https://github.com/bitshares/bitshares-core/blob/bf263b1db47a91ea9748118df1abb5a22cafc40b/libraries/chain/asset_object.cpp#L77

Reply:
https://github.com/bitshares/bitshares-core/blob/bf263b1db47a91ea9748118df1abb5a22cafc40b/libraries/chain/asset_object.cpp#L50-L58

Reply:
So actually you need to update the time.

Reply:
I think the logic is wrong in `update_median_feeds`. Since `null` price is always valid, returning `null` as median is allowed. But perhaps comparison will fail if `null` is involved? If this is the case, it's a serious bug.

Reply:
OK. It's not too serious since `price() < price()` always return `false`. So worst case is `price()` be returned. I think that's fine. Need to review more code though.